docs/docker.md

```diff
@@ -0,0 +1,101 @@
+### Docker
+
+#### Building the image
+
+```
+docker build -t github/backup-utils .
+```
+
+#### Setting configuration options at runtime
+The `backup.config` file is dynamically populated at runtime with all `GHE_` environment variables that are part of the run command or Docker environment:
+
+```
+$ docker run -it -e "GHE_HOSTNAME=hostname" \
+-e "GHE_DATA_DIR=/data" \
+-e "GHE_EXTRA_SSH_OPTS=-i /ghe-ssh/id_rsa -o UserKnownHostsFile=/ghe-ssh/known_hosts \
+-e "GHE_NUM_SNAPSHOTS=15" \
+-v ghe-backup-data:/data \
+-v "$HOME/.ssh/known_hosts:/ghe-ssh/known_hosts" \
+-v "$HOME/.ssh/id_rsa:/ghe-ssh/id_rsa" \
+--rm \
+github/backup-utils ghe-backup
+```
+
+It is also possible to specify a `-e GHE_BACKUP_CONFIG` flag and volume mount in a local `backup.config` file rather than specify the variables individually at run time:
+
+```
+$ docker run -it  -e "GHE_BACKUP_CONFIG=/mnt/backup.config" \
+-v ghe-backup-data:/data \
+-v "$HOME/.ssh/known_hosts:/ghe-ssh/known_hosts" \
+-v "$HOME/.ssh/id_rsa:/ghe-ssh/id_rsa" \
+-v $HOME/backup-utils/backup.config:/mnt/backup.config \
+--rm \
+github/backup-utils ghe-backup
+```
+
+#### SSH Keys
+
+A SSH private key that has been added to the GitHub Enterprise [Management Console for administrative SSH access](https://help.github.com/enterprise/admin/guides/installation/administrative-shell-ssh-access/) needs to be mounted into the container from the host system. It is also recommended to mount a SSH `.ssh/known_hosts` file into the container.
+
+```
+$ docker run -it -e "GHE_HOSTNAME=hostname" \
+-e "GHE_DATA_DIR=/data" \
+-e "GHE_EXTRA_SSH_OPTS=-i /ghe-ssh/id_rsa -o UserKnownHostsFile=/ghe-ssh/known_hosts \
+-v ghe-backup-data:/data \
+-v "$HOME/.ssh/known_hosts:/ghe-ssh/known_hosts" \
+-v "$HOME/.ssh/id_rsa:/ghe-ssh/id_rsa" \
+--rm \
+github/backup-utils ghe-backup
+```
+
```

Comment:
Additionally, I note we're making assumptions here that the customer knows how to handle forwarding the SSH agent to the container or otherwise handle SSH keys with passphrases. Do we want to make some sort of statement about it here or provide a tip?

Reply:
Good point. I had only been testing with passwordless keys. Let me do a little testing, I think we can get the host's SSH agent into the container.

Reply:
Looks like it will work on Linux Docker hosts, but not on Docker for Mac just yet https://github.com/docker/for-mac/issues/483.

I suppose adding a statement saying something about this limitation will work for now.

Reply:
Added a section on using `ssh-agent` - Seems to work pretty well on a Linux host using a SSH key protected by a passphrase.

Reply:
Addid a bit about how to mount `ssh-agent` into the container, which should take care of this.